### ISSUE TITLE
site sends incorrectly encoded emails on PHP 8 only

### ISSUE THREAD

**Issue by ananelson** [friendica/friendica#11517]
Hello,

I'm a first-time friendica site admin, and I have just installed a new site.

I am running into an issue where site emails work fine when using PHP 7.4, but when I switch to PHP8 the emails are being incorrectly encoded, so what is seen in the email when viewed in a client is base64 text in various mime type sections, and not the rendered content. Here is a screenshot from my email client showing a fragment of a password reset email:

<img width="263" alt="image" src="https://user-images.githubusercontent.com/14220/168493064-564aa998-0d32-4083-ad27-4cbba86889f7.png">

Is there a PHP setting I need to change, which might have a different default between PHP 7 and 8, which could be causing this issue? Or any other ideas?

**Comment by MrPetovan** [comment]
Thank you for your report, what version of PHP 8 are you using?

**Comment by ananelson** [comment]
Currently running 8.1.5. I tried with 8.0.? as well and got the same behavior with both.

**Comment by MrPetovan** [comment]
Thank you for the elaboration! I do not have access to PHP 8 on my server, but this is a good start. Do you mind sharing the email raw text with us for additional clues?

**Comment by ananelson** [comment]
I sent to your email.

**Comment by MrPetovan** [comment]
Can you please send me an email raw message with PHP 7 so that I can compare? So far I couldn't spot any obvious issues with the part of the email we are generating. I was also wrong, the core emails aren't handled by a third-party library, we put the mail body together ourselves, which isn't ideal for the reason you uncovered in this issue.

**Comment by ananelson** [comment]
I noticed that parts of the message were separated with "\r\n" and others with "\n" so I standardized to "\r\n" and that resolved the issue for me. It's probably easier for you to just make the same changes but I can make a PR if that would be helpful. (And yes as you noted, probably using a library is best in the long term.)

**Comment by MrPetovan** [comment]
Yes, please submit a PR, I didn't take the time to dive into this issue further so I'll gladly take your fix. I also noticed unexpected line feeds in the raw message but I wasn't sure how to fix them.

**Comment by ananelson** [comment]
Will do! Which branch would you like a PR against?

**Comment by MrPetovan** [comment]
`2022.05-rc` please. Don't worry too much about it when you submit your PR, we can switch branch easily.

**Comment by MrPetovan** [comment]
@ananelson Did you just change all the `\n` to `\r\n` in `Emailer.php` or was there something else?

**Comment by ananelson** [comment]
There were 2 places I think - sorry I haven't been able to do much coding lately but I should be able to get back to this later this week.

**Comment by MrPetovan** [comment]
Thanks, so far your hints haven't been enough for me to be sure I have the right fix for a problem I can't reproduce without a PHP 8.1 install. I can wait for you to be available.

**Comment by MrPetovan** [comment]
Hi @ananelson , were you able to look into this? A diff would really help since I can't seem to be able to reproduce the issue myself.
